packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/RealmAnyNestedCollectionTests.kt

```diff
@@ -0,0 +1,788 @@
+                nestedList[0]
+            }
+
+            // Updating to a new list
+            instance.value = realmAnyListOf(7)
+            // Accessing original orphaned list return 7 from the new instance again, but expected ILLEGAL_STATE_EXCEPTION["List is no longer valid"]
+            assertFailsWithMessage<IllegalStateException>("List is no longer valid") {
+                nestedList[0]
+            }
+        }
+    }
+
+    @Test
+    fun query_ThrowsOnNestedCollectionArguments() {
+        assertFailsWithMessage<IllegalArgumentException>("Invalid query argument: Cannot pass unmanaged collections as input argument") {
+            realm.query<JsonStyleRealmObject>("value == $0", RealmAny.create(realmSetOf()))
+        }
+        assertFailsWithMessage<IllegalArgumentException>("Invalid query argument: Cannot pass unmanaged collections as input argument") {
+            realm.query<JsonStyleRealmObject>("value == $0", RealmAny.create(realmListOf()))
+        }
+        assertFailsWithMessage<IllegalArgumentException>("Invalid query argument: Cannot pass unmanaged collections as input argument") {
+            realm.query<JsonStyleRealmObject>("value == $0", RealmAny.create(realmDictionaryOf()))
+        }
+    }
+
+    @Test
+    fun query() = runBlocking<Unit> {
+        realm.write {
+            copyToRealm(
+                JsonStyleRealmObject().apply {
+                    id = "SET"
+                    value = realmAnySetOf(1, 2, 3)
+                }
+            )
+            copyToRealm(
+                JsonStyleRealmObject().apply {
+                    id = "LIST"
+                    value = realmAnyListOf(4, 5, 6)
+                }
+            )
+            copyToRealm(
+                JsonStyleRealmObject().apply {
+                    id = "DICT"
+                    value = realmAnyDictionaryOf(
+                        "key1" to 7,
+                        "key2" to 8,
+                        "key3" to 9,
+                    )
+                }
+            )
+            copyToRealm(
+                JsonStyleRealmObject().apply {
+                    id = "EMBEDDED"
+                    value = realmAnyListOf(
+                        setOf(1, 2, 3),
+                        listOf(4, 5, 6),
+                        mapOf(
+                            "key1" to 7,
+                            "key2" to 8,
+                            "key3" to listOf(9),
+                        )
+                    )
+                }
+            )
+        }
+
+        assertEquals(4, realm.query<JsonStyleRealmObject>().find().size)
+
+        // Matching lists
+        realm.query<JsonStyleRealmObject>("value[0] == 4").find().single().run {
+            assertEquals("LIST", id)
+        }
+        realm.query<JsonStyleRealmObject>("value[*] == 4").find().single().run {
+            assertEquals("LIST", id)
+        }
+
+        // Matching dictionaries
```

Comment:
Yes. `"value[*] == 7"` would match an object with a dictionary with any key that has a value `7`. But it will also match an object with a list containing `7`. I added the dictionary case in the test.